eve/buildplan/test_core.py

```diff
@@ -0,0 +1,236 @@
+                        ),
+                    ]
+                ),
+            ),
+            (
+                MULTI_STAGES,
+                OrderedDict(
+                    [
+                        ("version", "0.2"),
+                        ("branches", OrderedDict()),
+                        (
+                            "stages",
+                            OrderedDict(
+                                [
+                                    (MULTI_STAGES.name, MULTI_STAGES.dump()),
+                                    (SINGLE_STAGE.name, SINGLE_STAGE.dump()),
+                                ]
+                            ),
+                        ),
+                    ]
+                ),
+            ),
+        ),
+        ids=("empty", "no-branch", "single-stage", "multi-stages"),
+    )
+    def test_project(self, stage, expected):
+        project = core.Project()
+
+        if stage is not None:
+            project.add(stage)
+
+        assert project.dump() == expected
+
+    @pytest.mark.parametrize(
+        "stage,expected",
+        (
+            (
+                SINGLE_STAGE,
+                OrderedDict(
+                    [("worker", core.LocalWorker().dump()), ("steps", [])]
+                ),
+            ),
+            (
+                MULTI_STAGES,
+                OrderedDict(
+                    [
+                        ("worker", core.LocalWorker().dump()),
+                        ("steps", [TRIGGER_STEP.dump()]),
+                    ]
+                ),
+            ),
+        ),
+        ids=("single", "multiple"),
+    )
+    def test_stage(self, stage, expected):
+        assert stage.dump() == expected
+
+    @pytest.mark.parametrize(
+        "step,expected",
+        (
+            (
+                TRIGGER_STEP,
+                OrderedDict(
+                    [
+                        (
+                            "TriggerStages",
+                            OrderedDict(
+                                [
+                                    ("name", "Trigger another stage"),
+                                    ("stage_names", ["single-stage"]),
+                                ]
+                            ),
+                        )
+                    ]
+                ),
+            ),
+        ),
+        ids=("trigger-stages",),
+    )
+    def test_step(self, step, expected):
+        assert step.dump() == expected
+
+    @pytest.mark.parametrize(
+        "worker,expected",
+        ((core.LocalWorker(), OrderedDict([("type", "local")])),),
+        ids=("local",),
+    )
+    def test_worker(self, worker, expected):
+        assert worker.dump() == expected
+
+
+class TestProject:
+    def test_add(self):
+        project = core.Project()
+
+        stage = core.Stage(name="simple", worker=core.LocalWorker(), steps=[])
+
+        project.add(stage)
+        assert sum(stage.name == s.name for s in project.stages) == 1
```

Comment:
maybe an `assert len(filter(…, project.stages)) == 1` (or using list comprehension) would be more explicit that relying on the fact that `True` is automagically casted to `1`